### PR TITLE
Fix mismatch between name node_name in FeatureConfig mapping.

### PR DESCRIPTION
**Pull request by mbrette**
This mismatch led to different prediction between java/scala and python

### CODE REVIEW

File: jvm/ml4ir-inference/src/main/scala/ml4ir/inference/tensorflow/data/ModelFeaturesConfig.scala [hunk 1]
```diff
@@ -35,13 +35,14 @@ case class ModelFeaturesConfig(@JsonProperty("rank") initialRank: FeatureConfig,
...
-  * @param nodeName the Tensorflow input graph node name to wire this feature into
+  * @param name the protobuf key name representing the input. Tensorflow will wire the value into the corresponding node
+  *             name
   * @param dTypeString string | int64 | float
   * @param servingConfig {@see ServingConfig} below
   * @param tfRecordType context | sequence (latter only for {@see SequenceExample} features)
   */
 @JsonIgnoreProperties(ignoreUnknown = true)
-case class FeatureConfig(@JsonProperty("node_name") nodeName: String,
+case class FeatureConfig(@JsonProperty("name") name: String,
```
Comment (balikasg) [inline jvm/ml4ir-inference/src/main/scala/ml4ir/inference/tensorflow/data/ModelFeaturesConfig.scala:45]: agreed. This raises the question if we need node_name and why. Same question for serving_info.name. 
Right now we have 3 names, I am wondering if we can live with 2.

**Review by balikasg — DISMISSED** [review]
great!!

File: jvm/ml4ir-inference/src/main/scala/ml4ir/inference/tensorflow/data/ModelFeaturesConfig.scala (hunk 1 above)
Comment (mbrette) [reply jvm/ml4ir-inference/src/main/scala/ml4ir/inference/tensorflow/data/ModelFeaturesConfig.scala:45]: I think we need node_name (it has to be unique, and it is visible in the graph, so useful for debug to have meaningful name).
But maybe we could get rid of serving_info.name.

Comment (mbrette) [reply jvm/ml4ir-inference/src/main/scala/ml4ir/inference/tensorflow/data/ModelFeaturesConfig.scala:45]: BTW, I double-checked that 'name' is not used as a key to a map of feature info. In the python code, the features are always manipulated as a list. (so 'name' does not have to be unique)

File: jvm/ml4ir-inference/src/test/resources/classification/feature_config_with_same_name.yaml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        max_length: 20
+    preprocessing_info:
+      - fn: preprocess_text
+        args:
+          remove_punctuation: true
+          to_lower: true
+    serving_info:
+      name: query_text
+      required: true
+    default_value: ""
```
Comment (Ullimague) [inline jvm/ml4ir-inference/src/test/resources/classification/feature_config_with_same_name.yaml:59]: minor: default_value should be under serving_info.

File: jvm/ml4ir-inference/src/test/resources/classification/feature_config_with_same_name.yaml [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+      fn: categorical_embedding_with_vocabulary_file
+      args:
+        vocabulary_file: ml4ir/applications/classification/tests/data/configs/vocabulary/domain_id.csv
+        embedding_size: 64
+        default_value: -1
+        num_oov_buckets: 1
+    serving_info:
+      name: domain_id
+      required: true
+    default_value: ""
```
Comment (Ullimague) [inline jvm/ml4ir-inference/src/test/resources/classification/feature_config_with_same_name.yaml:119]: minor: default_value should be under serving_info.

File: jvm/ml4ir-inference/src/test/resources/classification/feature_config_with_same_name.yaml [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+      args:
+        vocabulary_file: ml4ir/applications/classification/tests/data/configs/vocabulary/entity_id.csv
+        embedding_size: 64
+        encoding_size: 64
+        num_oov_buckets: 1
+        max_length: 20
+    serving_info:
+      name: user_context
+      required: true
+    default_value: ""
```
Comment (Ullimague) [inline jvm/ml4ir-inference/src/test/resources/classification/feature_config_with_same_name.yaml:154]: minor: default_value should be under serving_info.

File: jvm/ml4ir-inference/src/test/resources/classification/feature_config_with_same_name.yaml [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    - fn: one_hot_vectorize_label
+  feature_layer_info:
+    type: numeric
+    fn: categorical_indicator_with_vocabulary_file
+    args:
+      vocabulary_file: ml4ir/applications/classification/tests/data/configs/vocabulary/entity_id.csv
+  serving_info:
+    name: entity_id
+    required: false
+  default_value: ""
```
Comment (Ullimague) [inline jvm/ml4ir-inference/src/test/resources/classification/feature_config_with_same_name.yaml:34]: minor: default_value should be under serving_info.

Comment (jakemannix) [reply jvm/ml4ir-inference/src/test/resources/classification/feature_config_with_same_name.yaml]: addressed in my update to this branch, @Ullimague !

**Review by Ullimague — APPROVED** [review]

**Pull request merged**